### PR TITLE
plugins on stevedore

### PR DESCRIPTION
**Pull request by ch40s3k**

### CODE REVIEW

File: alamo_worker/plugins/__init__.py [hunk 1]
```diff
@@ -144,7 +106,11 @@ def dispatch(self, loop: 'ZMQEventLoop', payload: dict) -> Optional[dict]:
         :param dict payload:
         """
         _plugin_type = payload.get('type')
-        _plugin = self._plugins.get(_plugin_type)
+        _plugins = dict((x.name, x.obj) for x in self.runner)
```
Comment (deejay1) [inline alamo_worker/plugins/__init__.py]: why not a straight dict comprehension?

File: alamo_worker/plugins/__init__.py [hunk 2]
```diff
@@ -144,7 +106,11 @@ def dispatch(self, loop: 'ZMQEventLoop', payload: dict) -> Optional[dict]:
...
+        _plugin = None
+        for ext in _plugins.values():
+            if _plugin_type in ext.supported_types():
+                _plugin = ext
         if _plugin is None:
             logger.error('Could not find plugin ``{}``.'.format(_plugin_type))
```
Comment (deejay1) [inline alamo_worker/plugins/__init__.py]: fix the logging call when you're at it

File: alamo_worker/plugins/__init__.py [hunk 3]
```diff
@@ -72,68 +74,28 @@ class PluginManager(object):
...
-            return
-
-        if cls._type in self._classes:
-            logger.warn('Plugin "{}" already registered.'.format(cls))
-        else:
-            logger.info('Plugin "{}" registered successfully.'.format(
-                cls.__name__)
-            )
-            self._classes[cls._type] = cls
+        self.runner = NamedExtensionManager(
```
Comment (deejay1) [inline alamo_worker/plugins/__init__.py]: I wonder if https://docs.openstack.org/developer/stevedore/managers.html#namedispatchextensionmanager would fit here based on the check type?

File: setup.cfg [hunk 4]
```diff
@@ -23,3 +23,8 @@ source-dir = docs/source
 [entry_points]
 console_scripts =
   alamo-worker = alamo_worker.cli:main
+
+pl.allegro.tech.monitoring.alamo_worker.plugins =
+    http_check = alamo_worker.plugins.http_check:HttpCheckPlugin
+    graphite = alamo_worker.plugins.graphite:GraphitePlugin
+    druid = alamo_worker.plugins.druid:DruidPlugin
```
Comment (deejay1) [inline setup.cfg]: missing newline at end of file ;)

Comment (ch40s3k) [inline setup.cfg]: it is *.cfg not py

File: alamo_worker/plugins/__init__.py (hunk 2 above)
Comment (ch40s3k) [inline alamo_worker/plugins/__init__.py]: Done

File: alamo_worker/plugins/__init__.py (hunk 1 above)
Comment (ch40s3k) [inline alamo_worker/plugins/__init__.py]: Done

File: alamo_worker/plugins/__init__.py [hunk 5]
```diff
@@ -72,68 +74,28 @@ class PluginManager(object):
...
-    def load(self, plugins):
+    def load(self):
         """Load and instantiate plugins.
 
         :param plugins: plugin list
         """
-        self._import_plugins(plugins)
-        self._instantiate_plugins(plugins)
+        for plug in pkg_resources.iter_entry_points(self.plugin_namespace):
+            logger.debug("Loading {}".format(plug))
```
Comment (ch40s3k) [inline alamo_worker/plugins/__init__.py]: Done

File: alamo_worker/plugins/__init__.py (hunk 3 above)
Comment (ch40s3k) [inline alamo_worker/plugins/__init__.py]: Need to test it.

Comment (ch40s3k) [inline alamo_worker/plugins/__init__.py]: It is good as it is right now.

File: alamo_worker/plugins/__init__.py [hunk 6]
```diff
@@ -72,68 +74,28 @@ class PluginManager(object):
     """
     _plugins = {}
```
Comment (pnuckowski) [inline alamo_worker/plugins/__init__.py]: Is it used? If not it could be dropped.

File: alamo_worker/plugins/druid.py [hunk 7]
```diff
@@ -59,6 +58,9 @@ def __init__(self):
     def init(self, *args):
         super().init(*args)
 
+    def supported_types(self):
+        return self._type
```
Comment (pnuckowski) [inline alamo_worker/plugins/druid.py]: return type list? `[self._type]`

File: alamo_worker/plugins/http_check.py [hunk 8]
```diff
@@ -76,6 +75,9 @@ def _prepare_consul(self, session, url: str, env: str=None):
             except (AttributeError, KeyError):
                 raise EnvironmentFetchException
 
+    def supported_types(self):
+        return self._type
```
Comment (pnuckowski) [inline alamo_worker/plugins/http_check.py]: as list ?

File: alamo_worker/plugins/tests/test_plugin_manager.py [hunk 9]
```diff
@@ -1,22 +1,28 @@
...
 
 
-@plugin
 class TestPlugin(BasePlugin):
     _type = 'test'
 
-    def execute(self, *args):
-        pass
+    def supported_types(self):
+        return self._type
```
Comment (pnuckowski) [inline alamo_worker/plugins/tests/test_plugin_manager.py]: ditto

File: alamo_worker/plugins/http_check.py (hunk 8 above)
Comment (pnuckowski) [inline alamo_worker/plugins/http_check.py]: maybe typehint for new/update methods?

**Pull request merged**